### PR TITLE
nuke clang-wrapper and use gcc-wrapper with clang

### PR DESCRIPTION
**Pull request by gridaphobe**
This will force a rebuild of the world on linux unfortunately, but it seems to work so far. I'm still rebuilding my environment though.

/cc @shlevy @jwiegley @joelteon @copumpkin

### CODE REVIEW

**Comment by jwiegley** [comment]
@gridaphobe Let me know here when you have it building on your machine, and I'll fire up tests in my VMs.

**Comment by copumpkin** [comment]
Great work! Can we rename it to just `cc-wrapper` going forward? Seems consistent with the other change we pushed upstream.

**Comment by gridaphobe** [comment]
I'd support calling it `cc-wrapper` but i don't really want to rehash the whole "does gcc mean gcc or gcc-compatible" discussion :/

**Comment by copumpkin** [comment]
Well, we already got the gcc->cc rename into mainline, so this seems like more of the same. If it's general enough to be able to wrap gcc and clang, and they're both called `cc` elsewhere, and we assign the result of the wrapper to an attribute called `cc`, it seems sensible to just call this a `cc-wrapper`.

Compare `cc = wrapGcc { gcc = clang }` (wtf!) to `cc = wrapCc { cc = clang }`!

File: pkgs/build-support/gcc-wrapper/default.nix [hunk 1]
```diff
@@ -204,11 +212,11 @@ stdenv.mkDerivation {
       cp -p ${./utils.sh} $out/nix-support/utils.sh
 
       if [ -e $out/bin/clang ]; then
-        echo 'export CC; : ''${CC:=clang}' >> $out/nix-support/setup-hook
+        echo 'export CC=clang' >> $out/nix-support/setup-hook
```
Comment (shlevy) [inline pkgs/build-support/gcc-wrapper/default.nix]: On master, we have `CC=cc` and `CXX=c++`. Can we just stick with that?

File: pkgs/build-support/gcc-wrapper/gcc-wrapper.sh [hunk 2]
```diff
@@ -79,6 +79,18 @@ if [ "$NIX_ENFORCE_PURITY" = 1 -a -n "$NIX_STORE" ]; then
     params=("${rest[@]}")
 fi
 
+if test -n "@libcxx@"; then
```
Comment (shlevy) [inline pkgs/build-support/gcc-wrapper/gcc-wrapper.sh]: In theory is it possible to use `gcc` + `libc++`? Has anyone tested that combination with this wrapper?

**Comment by shlevy** [comment]
Minor comments aside, this looks good. Moving to `cc-wrapper` and opening this against upstream seems like the next step, let me know if you'd rather I did that.

File: pkgs/build-support/gcc-wrapper/default.nix (hunk 1 above)
Comment (gridaphobe) [inline pkgs/build-support/gcc-wrapper/default.nix]: Ah yes, the `setup-hook` here sets `CC=gcc` and `CXX=g++`, but if we can just use `cc` and `c++` instead that'd be better :)

File: pkgs/build-support/gcc-wrapper/gcc-wrapper.sh (hunk 2 above)
Comment (gridaphobe) [inline pkgs/build-support/gcc-wrapper/gcc-wrapper.sh]: I don't know, but perhaps @copumpkin or @joelteon do, I think they were the ones who added `libc++`.

**Comment by gridaphobe** [comment]
Oops the mode-change was an accident, emacs being too clever for its own good :)

**Comment by gridaphobe** [comment]
@shlevy ok I did the rename to cc-wrapper. Do you want to take another quick look before sending it upstream? (In particular I had to touch `stdenv/linux`, which I've never looked at before)

**Comment by shlevy** [comment]
@gridaphobe Overall looks good, I suspect there will be some more changes needed (e.g. `stdenv.cc.gcc` -> `stdenv.cc.cc`) but I think the best thing to do at this stage is let hydra have at it.

**Pull request merged**